src/components/Clipboard.react.js

```diff
@@ -0,0 +1,109 @@
+import React, {Component} from 'react'; // eslint-disable-line no-unused-vars
+import PropTypes from 'prop-types';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faCopy, faCheckCircle} from '@fortawesome/free-regular-svg-icons';
+
+/**
+ * The Clipboard component copies text to the clipboard
+ */
+
+export default class Clipboard extends React.Component {
+    constructor(props) {
+        super(props);
+        this.copyToClipboard = this.copyToClipboard.bind(this);
+        this.copySuccess = this.copySuccess.bind(this);
+        this.getText = this.getText.bind(this);
+        this.state = {
+            copied: false,
+            hasNavigator: true,
+        };
+    }
+
+    copySuccess() {
+        const showCopiedIcon = 1000;
+        this.setState({copied: true}, () => {
+            setTimeout(() => {
+                this.setState({copied: false});
+            }, showCopiedIcon);
+        });
+    }
+
+    getText() {
+        // get the inner text.  If none, use the content of the value param
+        var text = document.getElementById(this.props.target_id).innerText;
+        if (text === undefined || text === '') {
+            text = document.getElementById(this.props.target_id).value;
+            text = text === undefined ? '' : text;
+        }
+        return text;
+    }
+
+    copyToClipboard() {
+        const text = this.getText();
+        navigator.clipboard.writeText(text).then(
+            () => {
+                this.copySuccess();
+            },
+            function() {
+                alert('copy error');
+            }
+        );
+    }
+
+    componentDidMount() {
+        if (!navigator.clipboard) {
+            this.setState({hasNavigator: false});
+            alert('Copy to clipboard not available with this browser');
```

Comment:
I feel like an `alert` is too much in this case. Most of the time this component is just a convenience to the user, without it they can select and cmd/ctrl-C. So I wouldn't interrupt their experience with an alert. `console.warn` feels sufficient to me, so the developer will know what happened but as far as users are concerned the icon silently fails to render.

Also I don't think `hasNavigator` needs to be `state` (nor is it quite the right name...) - I bet we could make a module-level `const clipboardAPI = navigator.clipboard`, test for it in the constructor `if (!clipboardAPI) { console.warn(...) }` and even access it that way `clipboardAPI.writeText(text)`

Reply:
Good points - I included both suggestions

Reply:
Hey @AnnMarieW 

Why I am getting "Copy to clipboard not available with this browser" on dash 1.20.0

Reply:
Hi @AltafHussain4748  
The `Clipboard` component was not added until V1.21.0, so if you are trying to use it with V1.20.0, I would expect you to see an error like:
`module 'dash_core_components' has no attribute 'Clipboard'`

If you are using it with V1.21.0 and see the message: `Copy to clipboard not available with this browser`, it's because you are using an older browser that doesn't support the Clipboard API.  See the  browser compatibility list [here](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API).